.github/workflows/image-update.yaml

```diff
@@ -10,15 +10,50 @@ jobs:
     runs-on: ubuntu-18.04
     steps:
       - uses: actions/checkout@v1
-      - run: make -f Makefile.docker build-image
+      # Login to both registries
       - run: echo ${DOCKER_HUB_PASSWORD} | docker login --username weaveworkseksctlci --password-stdin
         env:
           DOCKER_HUB_PASSWORD: ${{ secrets.DOCKER_HUB_PASSWORD }}
-      - run: make -f Makefile.docker push-build-image
+      - name: Update the build image, push it Docker Hub and update manifest in the repo
+        env:
+          DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+        run: |
+          set -x
+
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          # Update image manifest and push new image to registry only when needed
+          if ! make -f Makefile.docker check-build-image-manifest-up-to-date ; then
+            # Commit changes to the manifest
+            git config --local user.email "weaveworksbot@users.noreply.github.com"
+            git config --local user.name "Weaveworks Bot"
+            make -f Makefile.docker update-build-image-manifest
+
+            # Build the image and push it to Docker Hub registry
+            make -f Makefile.docker build-image
+            make -f Makefile.docker push-build-image
+
+            # Configure private key
+            mkdir -p "${HOME}/.ssh"
+            ssh-keyscan -t rsa github.com > "${HOME}/.ssh/known_hosts"
+            echo "${DEPLOY_KEY}" > "${HOME}/.ssh/id_rsa"
```

Comment:
So I was hoping GitHub Actions would allow you to easily push back to a branch.

It was actually fairly easy to do:

git push `https://${{ github.actor }}:${{ secrete.GITHUB_TOKEN }}@github.com/weaveworks/eksctl <ref>`

However, as we attempt to update file in `.github/workflows/`, the commits get rejected... so we have to use SSH, and for that a deploy key had to be configured.